protocol/x/accountplus/authenticator/subaccount_filter.go

```diff
@@ -0,0 +1,110 @@
+package authenticator
+
+import (
+	"strconv"
+	"strings"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clobtypes "github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+)
+
+var _ Authenticator = &SubaccountFilter{}
+
+// SubaccountFilter filters incoming messages based on a whitelist of subaccount numbers.
+// It ensures that only messages with whitelisted subaccount numbers are allowed.
+type SubaccountFilter struct {
+	whitelist map[uint32]struct{}
```

Comment:
Any reason to keep `msgTypes []string` on the `MessageFilter`? I guess if we don't expect there to be a ton of msgType, then list is less overhead than `map`